pisek/cms/dataset.py

```diff
@@ -169,51 +169,52 @@ def add_judge(session: Session, files: FileCacher, env: Env, dataset: Dataset):
 def add_stubs(session: Session, files: FileCacher, env: Env, dataset: Dataset):
     config = env.config
 
-    if config.stub is None:
+    if not config.cms.stubs:
         return
 
     if config.task_type == TaskType.batch:
         stub_basename = "grader"
     elif config.task_type == TaskType.interactive:
         stub_basename = "stub"
 
-    directory, target_basename = path.split(config.stub.path)
-    directory = path.normpath(directory)
+    for stub in config.cms.stubs:
+        directory, target_name = path.split(stub.path)
+        directory = path.normpath(directory)
 
-    exts = set()
+        exts = set()
 
-    for filename in listdir(directory):
-        basename, ext = path.splitext(filename)
+        for filename in listdir(directory):
+            basename, ext = path.splitext(filename)
 
-        if basename != target_basename:
-            continue
+            if basename != target_name and filename != target_name:
+                continue
 
-        stub = files.put_file_from_path(
-            path.join(directory, filename),
-            f"{stub_basename}{ext} for {config.cms.name}",
-        )
-        session.add(
-            Manager(dataset=dataset, filename=f"{stub_basename}{ext}", digest=stub)
-        )
+            stub = files.put_file_from_path(
+                path.join(directory, filename),
+                f"{stub_basename}{ext} for {config.cms.name}",
+            )
+            session.add(
+                Manager(dataset=dataset, filename=f"{stub_basename}{ext}", digest=stub)
+            )
 
-        exts.add(ext)
+            exts.add(ext)
 
-    for ext, content in ERROR_STUBS.items():
-        if ext in exts:
-            continue
+        for ext, content in ERROR_STUBS.items():
+            if ext in exts:
+                continue
 
-        stub = files.put_file_content(
-            content.encode(), f"{stub_basename}{ext} for {config.cms.name}"
-        )
-        session.add(
-            Manager(dataset=dataset, filename=f"{stub_basename}{ext}", digest=stub)
-        )
+            stub = files.put_file_content(
+                content.encode(), f"{stub_basename}{ext} for {config.cms.name}"
+            )
+            session.add(
+                Manager(dataset=dataset, filename=f"{stub_basename}{ext}", digest=stub)
+            )
```

Comment:
When there are multiple stubs it doesn't make sense to add blank stubs for the complement of the languages supported by each of them.
Instead we should only upload blank stubs for languages not supported by any stub.